plutus-ledger-api/src/PlutusLedgerApi/Common/Versions.hs

```diff
@@ -173,10 +175,10 @@ and 'MajorProtocolVersion'?
 See Note [New builtins/language versions and protocol versions]
 -}
 builtinsAvailableIn :: PlutusLedgerLanguage -> MajorProtocolVersion -> Set.Set DefaultFun
-builtinsAvailableIn thisLv thisPv = fold $ Map.elems $
-    Map.takeWhileAntitone builtinAvailableIn builtinsIntroducedIn
+builtinsAvailableIn thisLv thisPv = fold $
+    Map.filterWithKey (const . alreadyIntroduced) builtinsIntroducedIn
     where
-      builtinAvailableIn :: (PlutusLedgerLanguage, MajorProtocolVersion) -> Bool
-      builtinAvailableIn (introducedInLv,introducedInPv) =
+      alreadyIntroduced :: (PlutusLedgerLanguage, MajorProtocolVersion) -> Bool
+      alreadyIntroduced (introducedInLv,introducedInPv) =
```

Comment:
Thanks for doing that.

@kwxm you were going to take a look at it, did you manage to do it in the end?

Reply:
> @kwxm you were going to take a look at it, did you manage to do it in the end?

A look at what?

Reply:
At the idea of replacing `takeWhileAntitone` with `filterWithKey` here.

Reply:
I can see how the highlighted diff and my comment were really confusing. Sorry!

Reply:
See [this](https://input-output-rnd.slack.com/archives/C06KVSMQS5D/p1715011137501089) thread.

Reply:
TBH, this whole Versions module needs some rework.